core/src/Streamly/Internal/FileSystem/Path/Common.hs

```diff
@@ -269,63 +280,148 @@ hasDrive a
     | not (isAlpha (unsafeIndexChar 0 a)) = False
     | otherwise = True
 
--- | On windows, the path starts with a separator.
-isCurDriveRelativeLocation :: (Unbox a, Integral a) => Array a -> Bool
-isCurDriveRelativeLocation a =
+-- | A path that starts with a alphabet followed by a colon e.g. @C:...@.
+hasDrive :: (Unbox a, Integral a) => Array a -> Bool
+hasDrive a = Array.byteLength a >= 2 && unsafeHasDrive a
+
+-- | A path that contains only an alphabet followed by a colon e.g. @C:@.
+isDrive :: (Unbox a, Integral a) => Array a -> Bool
+isDrive a = Array.byteLength a == 2 && unsafeHasDrive a
+
+------------------------------------------------------------------------------
+-- Relative or Absolute
+------------------------------------------------------------------------------
+
+-- | A path relative to cur dir it is either @.@ or starts with @./@.
+isRelativeCurDir :: (Unbox a, Integral a) => Array a -> Bool
+isRelativeCurDir a
     -- Assuming the path is not empty.
-    isSeparator Windows (wordToChar (Array.getIndexUnsafe 0 a))
+    | wordToChar (Array.getIndexUnsafe 0 a) /= '.' = False
+    | Array.byteLength a < 2 = True
+    | otherwise = isSeparator Windows (wordToChar (Array.getIndexUnsafe 1 a))
 
--- | @C:\...@
-isLocationDrive :: (Unbox a, Integral a) => Array a -> Bool
-isLocationDrive a
-    | Array.byteLength a < 3 = False
-    -- Check colon first for quicker return
-    | unsafeIndexChar 1 a /= ':' = False
-    | not (isSeparator Windows (unsafeIndexChar 2 a)) = False
-    -- XXX If we found a colon anyway this cannot be a valid path unless it has
-    -- a drive prefix. colon is not a valid path character.
-    -- XXX check isAlpha perf
-    | not (isAlpha (unsafeIndexChar 0 a)) = False
-    | otherwise = True
+-- | The path starting with a separator. On Windows this is relative to current
+-- drive while on Posix this is absolute path as there is only one drive.
+isRelativeCurDrive :: (Unbox a, Integral a) => OS -> Array a -> Bool
```

Comment:
Can you add some examples for this?
`isRelativeCurDrive` is any path that starts with a seperator?

Reply:
"/home/user" is relative to current drive.